A53-xds-ignore-resource-deletion.md

```diff
@@ -0,0 +1,113 @@
+then it is not necessarily safe for the client to simply ignore these
+deletions.  This could lead to a situation where the control plane
+deploys a bad update, but the operators think everything is fine because
+the clients are still working, and then a problem is discovered days or
+even months later when the clients are restarted and fail to come back up.
+This could lead to a different type of outage than the one that this design
+is intended to address.
+
+Because we cannot know whether a given control plane has appropriate
+alerting for human operators when a client is requesting a resource that
+does not exist, we propose to make this behavior configurable in gRPC.
+
+### Related Proposals: 
+
+* [gRFC A27: xDS-Based Global Load Balancing](A27-xds-global-load-balancing.md)
+* [A30: xDS v3 Support](A30-xds-v3.md)
+* [A47: xDS Federation](A47-xds-federation.md)
+
+## Proposal
+
+The ability to ignore resource deletions from the xDS server will be
+configured using the server feature mechanism introduced in [gRFC
+A30](A30-xds-v3.md).  Note that, as per [gRFC A47](A47-xds-federation.md),
+a separate server feature list can be configured for each server in a
+federated configuration, so this feature can be enabled on a per-server
+basis.
+
+The new server feature will be the string `ignore_resource_deletion`; if
+present, this feature will be enabled.
+
+When this feature is enabled, the XdsClient will not invoke the
+watchers' `OnResourceDoesNotExist()` method when a resource is deleted,
+nor will it remove the existing resource value from its cache.
+
+Currently, this behavior will affect only LDS and CDS, because those are
+the only two resource types for which the state-of-the-world xDS protocol
+variant allows the server to indicate a deletion to a client.  However,
+in the future, if/when gRPC adds support for the incremental xDS
+protocol variant, this behavior would affect all resource types.
+
+The XdsClient will generate an ERROR log when a resource deletion is
+ignored and a subsequent INFO log message when the condition clears, either
+because the server once again started sending the resource or because
+the client has unsubscribed from the resource.  To implement this, the
+XdsClient will maintain a bit in the cache entry for each resource
+indicating whether a deletion has been ignored.  When the XdsClient
+receives an LDS or CDS update that does not include the resource, if the
+bit is not set, it will log the ERROR message and set the bit; when the
+XdsClient receives an LDS or CDS update that includes a resource for
+which the cache entry bit is set, it will log the INFO message and unset the
+bit; when the XdsClient unsubscribes from a resource, if the cache entry bit
+is set, it will log the INFO message before deleting the cache entry.
+
+### Temporary environment variable protection
+
+This behavior is not configured via remote interaction, so it does not
```

Comment:
Note: we do have to handle backward compatibility with bootstrap files, so this is an incomplete argument.

Reply:
No, old clients ignoring the field still doesn't mean it is safe. The only thing that would make it safe is if we don't think we need interop testing, unit testing is sufficient, and it will be implemented in a single PR. A client with this logic half-way implemented should not break if the bootstrap generator is updated.

Reply:
Yeah, I was making exactly those assumptions.  But you're right that this should be called out explicitly.

Done.